src/silentpayments.rs

```diff
@@ -0,0 +1,546 @@
+    fn as_c_ptr(&self) -> *const Self::Target { &self.0 }
+
+    /// Obtains a mutable pointer suitable for use with FFI functions.
+    fn as_mut_c_ptr(&mut self) -> *mut Self::Target { &mut self.0 }
+}
+
+impl fmt::Display for SilentpaymentsFoundOutput {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+
+        let pubkey = XOnlyPublicKey::from(self.0.output.clone());
+
+        let buffer_str = pubkey.serialize()
+            .iter()
+            .map(|byte| format!("{:02x}", byte))
+            .collect::<String>();
+
+        write!(f, "{}", buffer_str)
+    }
+}
+
+impl SilentpaymentsFoundOutput {
+    /// Create a new `SilentpaymentsFoundOutput` object from a ffi::SilentpaymentsFoundOutput.
+    pub fn empty() -> SilentpaymentsFoundOutput {
+        SilentpaymentsFoundOutput(ffi::SilentpaymentsFoundOutput::empty())
+    }
+}
+
+/// Output scan errors
+#[derive(Debug, Clone, Copy, Eq, PartialEq, PartialOrd, Ord, Hash)]
+pub enum OutputScanError {
+    /// Unexpected failures
+    Failure,
+}
+
+#[cfg(feature = "std")]
+impl std::error::Error for OutputScanError {}
+
+impl fmt::Display for OutputScanError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+        match self {
+            OutputScanError::Failure => write!(f, "Failed to scan outputs"),
+        }
+    }
+}
+
+/// Scan for Silent Payment transaction outputs.
+pub fn silentpayments_recipient_scan_outputs<C: Verification, L>(
+    secp: &Secp256k1<C>,
+    tx_outputs: &[&XOnlyPublicKey],
+    recipient_scan_key: &SecretKey,
+    public_data: &SilentpaymentsPublicData,
+    recipient_spend_pubkey: &PublicKey,
+    label_lookup: SilentpaymentsLabelLookupFunction,
+    label_context: L,
+) -> Result<Vec<SilentpaymentsFoundOutput>, OutputScanError>
+{
+
+    let cx = secp.ctx().as_ptr();
+
+    let n_tx_outputs = tx_outputs.len();
+
+    let mut out_found_output = vec![ffi::SilentpaymentsFoundOutput::empty(); n_tx_outputs];
+    let mut out_found_output_ptrs: Vec<_> = out_found_output.iter_mut().map(|k| k as *mut _).collect();
+
+    let mut n_found_outputs: usize = 0;
+
+    let res = unsafe {
+
+        let ffi_tx_outputs: &[*const ffi::XOnlyPublicKey] = transmute::<&[&XOnlyPublicKey], &[*const ffi::XOnlyPublicKey]>(tx_outputs);
+
+        secp256k1_silentpayments_recipient_scan_outputs(
+            cx,
+            out_found_output_ptrs.as_mut_c_ptr(),
+            &mut n_found_outputs,
+            ffi_tx_outputs.as_c_ptr(),
+            n_tx_outputs,
+            recipient_scan_key.as_c_ptr(),
+            public_data.as_c_ptr(),
+            recipient_spend_pubkey.as_c_ptr(),
+            label_lookup,
+            &label_context as *const L as *const c_void,
+        )
+    };
+
+    if res == 1 {
+
+        let mut result = vec![SilentpaymentsFoundOutput::empty(); n_found_outputs];
+        
+        for i in 0..n_found_outputs {
+            result[i] = SilentpaymentsFoundOutput(out_found_output[i]);
+
+        }
```

Comment:
Can be optimized to use `Vec::from_raw_parts` if `SilentpaymentsFoundOutput` is made `#[repr(transparent)]`.

Reply:
Thanks. Done in 6fbb0595e36f9acf4f24e20e43c7c790ee91f22e